File: mambo-protocol/src/main/scala/org/mambo/protocol/types/updater/ContentPart.scala





















// Generated on 05/03/2013 10:35:08
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait ContentPart extends TypeDeserializer {
  val typeId = 350

  protected trait Type extends BaseType with Serializable {
    val typeId = ContentPart.typeId

    val id: String
    val state: Byte

    def serialize(buf: Buffer) {
      buf.writeString(id)
      buf.writeByte(state)
    }

  }

  protected def create(id: String, state: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val id = buf.readString()
    val state = buf.readByte()

    create(id, state)
  }

}

object ContentPart extends ContentPart {
  case class TypeImpl(id: String, state: Byte) extends Type
  override def create(id: String, state: Byte) = TypeImpl(id, state)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(id: String, state: Byte) = create(id, state)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/ContentPart.java


// Generated on 05/08/2013 19:38:03
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ContentPart extends NetworkType {
    public static final short TYPE_ID = 350;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public String id;
    public byte state;
    
    public ContentPart() { }
    
    public ContentPart(String id, byte state) {
        this.id = id;
        this.state = state;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(id);
        buf.writeByte(state);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        id = buf.readString();
        state = buf.readByte();
        if (state < 0)
            throw new RuntimeException("Forbidden value on state = " + state + ", it doesn't respect the following condition : state < 0");
    }
    
}
